evalml/pipelines/pipeline_meta.py

```diff
@@ -16,18 +15,34 @@ def check_for_fit(cls, method):
             It raises an exception if `False` and calls and returns the wrapped method if `True`.
         """
         @wraps(method)
-        def _check_for_fit(self, X=None, y=None, objective=None):
+        def _check_for_fit(self, X=None, objective=None):
             klass = type(self).__name__
             if not self._is_fitted:
                 raise PipelineNotYetFittedError(f'This {klass} is not fitted yet. You must fit {klass} before calling {method.__name__}.')
-            elif X is None and y is None:
-                return method(self)
-            elif y is None:
+            if method.__name__ == 'predict_proba':
                 return method(self, X)
-            # For time series classification pipelines, predict will take X, y, objective
-            elif len(inspect.getfullargspec(method).args) == 4:
-                return method(self, X, y, objective)
-            # For other pipelines, predict will take X, y or X, objective
+            elif method.__name__ == 'predict':
+                return method(self, X, objective)
             else:
+                return method(self)
+        return _check_for_fit
+
+
+class TimeSeriesPipelineBaseMeta(PipelineBaseMeta):
+    """Metaclass that overrides creating a new time series pipeline by wrapping methods with validators and setters"""
+
+    @classmethod
+    def check_for_fit(cls, method):
+        """`check_for_fit` wraps a method that validates if `self._is_fitted` is `True`.
+            It raises an exception if `False` and calls and returns the wrapped method if `True`.
+        """
+        @wraps(method)
+        def _check_for_fit(self, X=None, y=None, objective=None):
+            klass = type(self).__name__
+            if not self._is_fitted:
+                raise PipelineNotYetFittedError(f'This {klass} is not fitted yet. You must fit {klass} before calling {method.__name__}.')
+            if method.__name__ == 'predict_proba':
                 return method(self, X, y)
+            elif method.__name__ == 'predict':
+                return method(self, X, y, objective)
```

Comment:
Ah, interestingly enough, the feature importance case is covered by `PipelineBaseMeta`, not `TimeSeriesPipelineBaseMeta`, so the `else` case is never reached if I have it (codecov caught this). I think this is because the implementation for feature importance lies in `PipelineBase` and isn't overridden by time series pipelines (confirmed via overriding time series pipelines' feature importance).